app/uk/gov/hmrc/customs/emailfrontend/controllers/WhatIsYourEmailController.scala

```diff
@@ -47,17 +51,21 @@ class WhatIsYourEmailController @Inject()(actions: Actions, view: what_is_your_e
     }
   }
 
-  def create: Action[AnyContent] = actions.auth { implicit request =>
-    Ok(view(emailForm))
+  def create: Action[AnyContent] = (actions.auth andThen actions.eori).async { implicit request =>
+    subscriptionDisplayConnector.subscriptionDisplay(Eori(request.eori.id)).map {
+      case SubscriptionDisplayResponse(Some(email)) => Ok(view(emailForm, email))
+    }
   }
 
-  def submit: Action[AnyContent] = actions.auth.async { implicit request =>
+  def submit: Action[AnyContent] = (actions.auth andThen actions.eori).async { implicit request =>
     emailForm.bindFromRequest.fold(
       formWithErrors => {
-        Future.successful(BadRequest(view(formWithErrors)))
+        subscriptionDisplayConnector.subscriptionDisplay(Eori(request.eori.id)).map {
```

Comment:
do we need to call subscription display (ETMP) again while submitting the form and just to show the errors on form page, doesn't feel great about this @shaileshp0110